Makefile

```diff
@@ -245,6 +245,9 @@ CFLAGS     += $(ARCH_FLAGS) \
               -fdata-sections \
               -fno-common \
               -pedantic \
+              -Wno-stringop-truncation \
```

Comment:
Can you please post the a list of the warnings that are generated for a representative target (STM32F405?). This will give us an indication of how much clean up work will be required.